src/compiler/checker.ts

```diff
@@ -36284,6 +36290,15 @@ namespace ts {
             }
         }
 
+        function markJSDocLinkNameAsReferenced(name: EntityName | JSDocMemberName) {
+            if (isEntityName(name)) {
+                resolveEntityName(name, SymbolFlags.Type | SymbolFlags.Namespace | SymbolFlags.Value, /*ignoreErrors*/ true, /*dontResolveAlias*/ true);

```

Comment:
for the non-experimental version of this, `resolveEntityName` will need to add the ad-hoc lookup semantics from resolveJSDocMemberName. I'm not sure what this will mean for supporting the syntax `Class#method`.

Reply:
> I'm not sure what this will mean for supporting the syntax Class#method

Do we need to set `isReferenced` on both `Class` and `method`? Or will only `Class` be sufficient? 

> for the non-experimental version of this, resolveEntityName will need to add the ad-hoc lookup semantics from resolveJSDocMemberName.

Do I need to use `resolveJSDocMemberName` directly in `resolveEntityName`? I used `resolveEntityName` for two reasons: 1. I think we can skip resolving `"prototype"` for this case (it looks like ts doesn't handle unused prototype members), 2. Second reason related to the first one is `ignoreErrors`, extending `resolveJSDocMemberName` with new argument `ignoreErrors ` and keep checking `"prototype"` seems useless. What do you think?

Reply:
> Do we need to set isReferenced on both Class and method? Or will only Class be sufficient?

I'm pretty sure `Class` will be sufficient although I could be wrong.

> Do I need to use resolveJSDocMemberName directly in resolveEntityName?

I think it might actually be the other way around, because after I think about it, I don't want the `@link` resolution rules to apply anywhere else. So the call you added to `resolveEntityName` should probably turn into a call to `resolveJSDocMemberName`, which itself will call `resolveEntityName`.
However, it may need some improvement to work right in that location. I don't have all the details of that code in my head, I just remember that it was pretty ad-hoc.